lib/iris/coords.py

```diff
@@ -2775,6 +2779,366 @@ def xml_element(self, doc):
+        if a different :attr:`cf_role` is needed.
+
+        """
+        return self._metadata_manager.cf_role
+
+    @property
+    def cf_role_element(self):
+        """
+        Derived from the connectivity's :attr:`cf_role` - the first part, e.g.
+        ``face`` in ``face_node_connectivity``. Refers to the elements
+        described by the structure of the connectivity's :attr:`indices` array.
+
+        """
+        return self._metadata_manager.cf_role.split("_")[0]
+
+    @property
+    def cf_role_indexed_element(self):
+        """
+        Derived from the connectivity's :attr:`cf_role` - the second part, e.g.
+        ``node`` in ``face_node_connectivity``. Refers to the elements indexed
+        by the values in the connectivity's :attr:`indices` array.
+
+        """
+        return self._metadata_manager.cf_role.split("_")[1]
+
+    @property
+    def start_index(self):
+        """
+        The base value of the connectivity's :attr:`indices` array; either
+        ``0`` or ``1``.
+        **Read-only** - to change value: use :meth:`switch_start_index`.
+
+        """
+        return self._metadata_manager.start_index
+
+    @property
+    def element_dim(self):
+        """
+        The dimension of the connectivity's :attr:`indices` array that details
+        individual elements.
+        **Read-only** - to change value: use :meth:`switch_element_dim`.
+
+        """
+        return self._metadata_manager.element_dim
+
+    @property
+    def indices(self):
+        """
+        The index values describing the topological relationship of the
+        connectivity, as a NumPy array. Masked points indicate an element
+        shorter than the longest element described in this array - unused
+        index 'slots' are masked.
+        **Read-only** - index values are only meaningful when combined with
+        an appropriate :attr:`cf_role`, :attr:`start_index` and
+        :attr:`element_dim`. A new :class:`Connectivity` must therefore be
+        defined if different indices are needed.
+
+        """
+        return self._values
+
+    @property
+    def element_lengths(self):
+        """
+        A NumPy array of the lengths of each element in the :attr:`element_dim`
+        of the connectivity's :attr:`indices` array, accounting for masks if
+        present.
+
+        """
+        lengths = self._lazy_element_lengths(self.indices, self.element_dim)
+        return lengths.compute()
+
+    @property
+    def has_equal_element_lengths(self):
+        """
+        A boolean stating whether all values from
+        :attr:`element_lengths` are equal. Provided
+        for lazy calculation without needing to realise :attr:`indices` or
+        :attr:`element_lengths`.
+
+        """
+        lengths = self._lazy_element_lengths(self.indices, self.element_dim)
+        has_equal = lengths.min() == lengths.max()
+        return has_equal.compute()
```

Comment:
Why do we need this behaviour? What's it's purpose?

This will be a massively expensive compute to perform.

Reply:
Since we aware of a specific use-case for UGRID that is strictly uniformly shaped faces (the LFRic cubesphere), I figured it would be useful to distinguish such configurations from the standard [2D flexible mesh](https://ugrid-conventions.github.io/ugrid-conventions/#2d-flexible-mesh-mixed-triangles-quadrilaterals-etc-topology), and I wanted to avoid any chance of realising the size of every face so it seemed simple to write this convenience.

>This will be a massively expensive compute to perform.

I had imagined that lazy computation would take care of this problem, but given you raised a concern I wrote a test and it used `1GB` for a couple of seconds. With that in mind I agree this definitely isn't worth offering.

Reply:
c04f2308cfab2a96ebb146a3034e468d397af06c